extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasConfig.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using MongoDB.Driver;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.KernelMemory.MongoDbAtlas;
+
+public class MongoDbAtlasConfig
+{
+    private MongoClient _client = null!;
+
+    public string ConnectionString { get; set; } = null!;
+
+    public string DatabaseName { get; set; } = null!;
+
+    public bool UseSingleCollectionForVectorSearch { get; set; } = false;
+
+    /// <summary>
+    /// A callback useful for tests, we need to wait that the index really indexed data after
+    /// we inserted into the collection so we need to wait a little bit.
+    /// </summary>
+    public Func<Task> AfterIndexCallbackAsync { get; private set; } = () => Task.CompletedTask;
+
+    public MongoDbAtlasConfig WithConnection(string mongoConnection)
+    {
+        this.ConnectionString = mongoConnection;
+        return this;
+    }
+
+    public MongoDbAtlasConfig WithDatabaseName(string databaseName)
+    {
+        this.DatabaseName = databaseName;
+        return this;
+    }
+
+    /// <summary>
+    /// If single collection for vector search is enabled, all the vectors will be stored in a single
+    /// collection, so we have only one search index in atlas. This can be useful to reduce index numbers.
+    /// </summary>
+    /// <param name="useSingleCollectionForVectorSearch"></param>
+    /// <returns></returns>
+    public MongoDbAtlasConfig WithSingleCollectionForVectorSearch(bool useSingleCollectionForVectorSearch)
+    {
+        this.UseSingleCollectionForVectorSearch = useSingleCollectionForVectorSearch;
+        return this;
+    }
+
+    public MongoDbAtlasConfig WithAfterIndexCallback(Func<Task> afterIndexCallback)
+    {
+        this.AfterIndexCallbackAsync = afterIndexCallback;
+        return this;
+    }
+
+    internal IMongoDatabase GetDatabase()
```

Comment:
this method seems a bit odd, e.g. having a client inside the config class. The code and the client instance could be moved to the `MongoDbAtlasBaseStorage` class.

Reply:
it could be done, but it is best practice to have a single Client for the entire application, so it is standard practice to leave the client in a single instance of a global class. Actually it would be also better (maybe in a subsequent change) to give the caller the ability to specify the client class from external code.


I've moved inside a dedicated class.

Reply:
the memory db object is a singleton, that should be sufficient no?